src/gateio_new_coins_announcements_bot/rotating_proxy.py

```diff
@@ -0,0 +1,105 @@
+from typing import Callable
+import requests
+import threading
+import time
+import itertools
+import socket
+import struct
+
+import gateio_new_coins_announcements_bot.globals as globals
+from gateio_new_coins_announcements_bot.logger import logger
+
+
+_proxy_list = {}
```

Comment:
`_proxy_list` can just a be list `[]` right? 
no need for a dict here.

Reply:
I used a dict to ensure unique proxies